sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/CosmosEncryptionAsyncContainer.java

```diff
@@ -0,0 +1,375 @@
+            + "EncryptionContainer.");
+
+
+        byte[] streamPayload = cosmosSerializerToStream(item);
+        CosmosItemRequestOptions finalRequestOptions = requestOptions;
+        return this.encryptionProcessor.encrypt(streamPayload).flatMap(encryptedPayload -> this.container.upsertItem(
+            encryptedPayload,
+            partitionKey,
+            finalRequestOptions)
+            .publishOn(encryptionScheduler).flatMap(cosmosItemResponse -> setByteArrayContent(cosmosItemResponse,
+                this.encryptionProcessor.decrypt(EncryptionModelBridgeInternal.getByteArrayContent(cosmosItemResponse)))
+                .map(bytes -> this.responseFactory.createItemResponse(cosmosItemResponse, (Class<T>) item.getClass()))));
+    }
+
+    /**
+     * replaces item and encrypts the requested fields
+     *
+     * @param item           the Cosmos item represented as a POJO or Cosmos item object.
+     * @param itemId         the item id.
+     * @param partitionKey   the partition key.
+     * @param requestOptions request option
+     * @param <T>            serialization class type
+     * @return result
+     */
+    @SuppressWarnings("unchecked")
+    public <T> Mono<CosmosItemResponse<T>> replaceItem(T item,
+                                                       String itemId,
+                                                       PartitionKey partitionKey,
+                                                       CosmosItemRequestOptions requestOptions) {
+        Preconditions.checkNotNull(item, "item");
+        if (requestOptions == null) {
+            requestOptions = new CosmosItemRequestOptions();
+        }
+
+        Preconditions.checkArgument(partitionKey != null, "partitionKey cannot be null for operations using "
+            + "EncryptionContainer.");
+
+
+        byte[] streamPayload = cosmosSerializerToStream(item);
+        CosmosItemRequestOptions finalRequestOptions = requestOptions;
+        return this.encryptionProcessor.encrypt(streamPayload).flatMap(encryptedPayload -> this.container.replaceItem(
+            encryptedPayload,
+            itemId,
+            partitionKey,
+            finalRequestOptions)
+            .publishOn(encryptionScheduler).flatMap(cosmosItemResponse -> setByteArrayContent(cosmosItemResponse,
+                this.encryptionProcessor.decrypt(EncryptionModelBridgeInternal.getByteArrayContent(cosmosItemResponse)))
+                .map(bytes -> this.responseFactory.createItemResponse(cosmosItemResponse, (Class<T>) item.getClass()))));
+    }
+
+    /**
+     * Reads item and decrypt the encrypted fields
+     *
+     * @param id             item id
+     * @param partitionKey   the partition key.
+     * @param requestOptions request options
+     * @param classType      deserialization class type
+     * @param <T>            type
+     * @return result
+     */
+    public <T> Mono<CosmosItemResponse<T>> readItem(String id,
```

Comment:
we previously had support for lazy decryption, which was very helpful on query.

is the support for lazy decryption removed?

Reply:
.NET also removed it in newer implementation, how ever they have both old and new (we can introduce in next PR when we iterate on preview on demand basis)